statistics/handle/update.go

```diff
@@ -756,14 +804,24 @@ func (h *Handle) HandleAutoAnalyze(is infoschema.InfoSchema) (analyzed bool) {
 				}
 				continue
 			}
+			// Only analyze the partition that has not been locked.
+			var partitionDefs []model.PartitionDefinition
+			for _, def := range pi.Definitions {
+				if _, ok := lockedTables[def.ID]; !ok {
+					partitionDefs = append(partitionDefs, def)
+				} else {
+					logutil.BgLogger().Info("skip analyze locked partition", zap.String("category", "stats"),
+						zap.String("db", db), zap.String("table", tblInfo.Name.O), zap.String("partition", def.Name.O))
```

Comment:
Similar to the "check table lock failed" log above, this will cause a lot of logs in some scenarios.

Reply:
But it's OK for me to keep this in this PR.

Reply:
Removed!